### PR TITLE
chore(ci): initial CI configuration for GitHub

### LINKED ISSUE #14: CI integration

### PR DESCRIPTION
**Pull request by Nithos**
Basic GitHub CI that is triggered on PR and direct pushes to main. Will be augmented or replaced with GitLab configuration once those runners are available.

resolves #14

### CODE REVIEW

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+name: Node.js CI
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (YohannParis) [inline .github/workflows/ci-test.yml:13]: Our DockerFile uses `node:18.8.0-slim` which I think is based on alpine. Should we use the same environment for CI and container?

File: .github/workflows/ci-test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js 18.4.0
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.4.0
+          cache: 'yarn'
```
Comment (YohannParis) [inline .github/workflows/ci-test.yml:20]: Same thing for Node.js and Yarn version with DockerFile

**Review by YohannParis — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
+		"postinstall": "husky install"
+	},
+	"lint-staged": {
+		"*.{ts,vue}": [
+			"eslint --fix",
+			"prettier --write"
+		]
+	},
+	"devDependencies": {
+		"@vue/eslint-config-airbnb-with-typescript": "^7.0.0",
```
[the remark this answers is not in the document]
Comment (Nithos) [reply package.json:24]: we did, this is an existing dependency that was present. I have not gone through and pinned these yet, will be part of the renovate job.

File: .github/workflows/ci-test.yml (hunk 1 above)
Comment (Nithos) [reply .github/workflows/ci-test.yml:13]: this is not the same, in GitHub workflows this defines the underlying system VM its running on and not the docker image. That being said just noticed that they are deprecating 18.x and we should probably be specific on the version so will upgrade to use v22

Comment (YohannParis) [reply .github/workflows/ci-test.yml]: I understand that they are not the same, I was just wondering if we should have the same configuration between DockerFile and GitHub Actions, just for saving ourselves headaches in the future.

Comment (Nithos) [reply .github/workflows/ci-test.yml]: don't have an option for alpine on GitHub as far as I can tell, hopefully once we move to gitlab runners we can make this exact

File: package.json [hunk 4]
```diff
@@ -1,31 +1,32 @@
...
+	"name": "terarium",
+	"packageManager": "yarn@3.2.2",
+	"private": true,
+	"workspaces": [
+		"packages/*"
+	],
+	"scripts": {
+		"build": "yarn workspaces foreach --parallel --interlaced --verbose --topological run build",
+		"dev": "yarn workspaces foreach --parallel --interlaced --verbose run dev",
+		"lint": "eslint . --ext .ts,.vue --fix",
```
Comment (jryu01) [inline package.json:11]: I wonder if we should have something like `lint:ci` or `prettier:ci` script that doesn't auto fix the linting/formatting errors for you to be used in the CI pipeline.

Comment (Nithos) [reply package.json:11]: good catch, will add them

Comment (Nithos) [reply package.json:11]: done

**Pull request merged**